DOCS: Add reference documentation for inspector views (#3990)
neos/neos-development-collection: Neos.Neos/Documentation/References/InspectorViewsReference/index.rst

                      'growthPercent' => 25,
                  ]
              ]
          ];
      }
  }

.. note:: The data returned from a data source that is used for a ``ColumnView`` need not adhere to a specific shape. The ``viewOptions`` configuration is used to extract data from arbitrary data shapes as you'll see below.

Example ``viewOptions``:

.. code-block:: yaml

  'Vendor.Site:MyCustomNodeType':
    # ...
    ui:
      inspector:
        views:
          columnViewExample:
            label: 'ColumnView Example'
            group: examples
            view: 'Neos.Neos/Inspector/Views/Data/ColumnView'
            viewOptions:
              dataSource: vendor-site-column-view
              hero:
                data: total.uniqueVisitors
                label: 'visitors'
              columns:
                -
                  data: thisWeek.uniqueVisitors
                  label: 'this week'
                -
                  data: thisMonth.uniqueVisitors
                  label: 'this month'
                -
                  data: thisMonth.growthPercent
                  label: '% growth'
    # ...

``viewOptions`` Reference:

``dataSource`` (required string)

>   Class name of class implementing ``Neos\Neos\Service\DataSource\DataSourceInterface``

Reviewer: ```suggestion
  Identifier for a DataSource

Reviewer 2: ```suggestion
  Identifier of a DataSource (given in the implementation class)
```

                  ],
              ]
          ];
      }
  }

.. note:: The data returned from a data source that is used for a ``TableView`` need not adhere to a specific shape (as long as it represents some sort of list). The ``viewOptions`` configuration is used to extract data from arbitrary data shapes as you'll see below.

Example ``viewOptions``:

.. code-block:: yaml

  'Vendor.Site:MyCustomNodeType':
    # ...
    ui:
      inspector:
        views:
          tableViewExample:
            label: 'TableView Example'
            group: examples
            view: 'Neos.Neos/Inspector/Views/Data/TableView'
            viewOptions:
              dataSource: vendor-site-table-view
              collection: rows
              columns:
                -
                  data: deviceCategory
                  iconMap:
                    desktop: icon-desktop
                    tablet: icon-tablet
                    smartphone: icon-mobile-phone
                -
                  data: uniqueVisitors
                  suffix: ' visitors'
                -
                  data: percentage
                  suffix: '%'
    # ...

``viewOptions`` Reference:

> ``dataSource`` (required string)

Reviewer: also here ;)

>   Class name of class implementing ``Neos\Neos\Service\DataSource\DataSourceInterface``

Reviewer 2: ```suggestion
  Identifier of a DataSource (given in the implementation class)
```

                          'uniqueVisitors' => 708
                      ],
                      [
                          'date' => new \DateTimeImmutable('2022-11-01'),
                          'uniqueVisitors' => 389
                      ],
                      [
                          'date' => new \DateTimeImmutable('2022-12-01'),
                          'uniqueVisitors' => 1027
                      ],
                  ],
              ]
          ];
      }
  }

.. note:: The data returned from a data source that is used for a ``TimeSeriesView`` need not adhere to a specific shape (as long as it includes a list of data associated with a time and date). The ``viewOptions`` configuration is used to extract data from arbitrary data shapes as you'll see below.

Example ``viewOptions``:

.. code-block:: yaml

  'Vendor.Site:MyCustomNodeType':
    # ...
    ui:
      inspector:
        views:
          timeSeriesViewExample:
            label: 'TimeSeriesView Example'
            group: examples
            view: 'Neos.Neos/Inspector/Views/Data/TimeSeriesView'
            viewOptions:
              subtitle: 'last month'

>               dataSource: vendor-site-time-series-view

Reviewer: And here ;)

Reviewer 2: Here it's fine, that is just an identifier…

              collection: rows
              series:
                timeData: date
                valueData: uniqueVisitors
              chart:
                selectedInterval: weeks
                yAxisFromZero: false
    # ...

``viewOptions`` Reference:

> ``dataSource`` (required string)

Reviewer: Aaand here :D

>   Class name of class implementing ``Neos\Neos\Service\DataSource\DataSourceInterface``

Reviewer 2: ```suggestion
  Identifier of a DataSource (given in the implementation class)
```
